src/main/java/com/hivemq/cli/commands/options/SubscribeOptions.java

```diff
@@ -48,37 +55,164 @@ public class SubscribeOptions {
+                        converter = Mqtt5UserPropertyConverter.class,
+                        description = "Options \"-up\" and \"--userProperty\" are legacy, please use \"--user-property\". They will be removed in a future version.")
+    private void userPropertiesLegacy(final @NotNull Mqtt5UserProperty @NotNull [] userPropertiesLegacy) {
+        //only show message once as this method is executed multiple times
+        if (!userPropertiesFromLegacyOption) {
+            deprecationWarnings.add(
+                    "Options \"-up\" and \"--userProperty\" are legacy, please use \"--user-property\". They will be removed in a future version.");
+        }
+
+        if (userPropertiesFromOption) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the user properties legacy options \"-up\" or \"--userProperty\" and the current \"--user-property\" is used. Please only use \"--user-property\" as the legacy options will be removed in a future version.");
+        }
+        userPropertiesFromLegacyOption = true;
+        userProperties = userPropertiesLegacy;
+    }
+
+    private @Nullable File outputFile = null;
+
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--output-to-file"},
                         description = "A file to which the received publish messages will be written")
-    private @Nullable File outputFile;
+    private void outputFile(final @NotNull File outputFile) {
+        if (this.outputFile != null) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the output file legacy options \"-of\" or \"--outputToFile\" and the current \"--output-to-file\" is used. Please only use \"--output-to-file\" as the legacy options will be removed in a future version.");
+        }
+        this.outputFile = outputFile;
+    }
+
+    @SuppressWarnings("unused")
+    @Deprecated(since = "4.34.0", forRemoval = true)
+    @CommandLine.Option(names = {"-of", "--outputToFile"},
+                        hidden = true,
+                        description = "Options \"-of\" and \"--outputToFile\" are legacy, please use \"--output-to-file\". They will be removed in a future version")
+    private void outputFileLegacy(final @NotNull File outputFileLegacy) {
+        deprecationWarnings.add(
+                "Options \"-of\" and \"--outputToFile\" are legacy, please use \"--output-to-file\". They will be removed in a future version.");
+
+        if (this.outputFile != null) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the output file legacy options \"-of\" or \"--outputToFile\" and the current \"--output-to-file\" is used. Please only use \"--output-to-file\" as the legacy options will be removed in a future version.");
+        }
+        this.outputFile = outputFileLegacy;
+    }
+
+    private boolean isPayloadEncodeToBase64 = false;
```

Comment:
```suggestion
    private boolean isPayloadEncodedToBase64 = false;
```

Reply:
Gave it some more thought. It is complicated 😄